src/pymordemos/elliptic2.py

```diff
@@ -75,16 +81,28 @@ def elliptic2_demo(args):
         name='2DProblem'
     )
 
+    if args['PRODUCT-COUNT'] and not args['--fv']:
+        # use a random parameter to construct an energy product
+        mu_bar = problem.parameter_space.sample_randomly(1)[0]
+    else:
+        mu_bar = None
+
     print('Discretize ...')
     discretizer = discretize_stationary_fv if args['--fv'] else discretize_stationary_cg
-    m, data = discretizer(problem, diameter=1. / args['N'])
+    if mu_bar is not None:
+        m, data = discretizer(problem, diameter=1. / args['N'], energy_product=mu_bar)
+    else:
+        m, data = discretizer(problem, diameter=1. / args['N'])
     print(data['grid'])
     print()
 
     print('Solve ...')
     U = m.solution_space.empty()
     for mu in problem.parameter_space.sample_uniformly(10):
         U.append(m.solve(mu))
+    if mu_bar is not None:
+        energy_norm_squared = m.products['energy'].apply2(U[-1], U[-1])
+        print('Energy norm of the last snapshot: ', np.sqrt(energy_norm_squared)[0][0])
```

Comment:
Why is the norm only computed in case of the energy norm?

Reply:
Because the energy product is only available in that case. And it is too much work to construct the energy product manually. This is how I wanted to show the difference between both discretizer calls. 
Anyways, in https://github.com/pymor/pymor/pull/1008/commits/80285ba4b3900fdb641726773f1de98938fc9aa4, I now also compute the H_1_0 semi norm being the closest 'brother' of the energy norm. In case the energy product is available, I compute both. What would you think would be prettier here?